### PR TITLE
Feature midterm

### PR DESCRIPTION
**Pull request by manuelmolinaf**

### CODE REVIEW

File: src/BoilerPlate/Game.cpp [hunk 1]
```diff
@@ -10,9 +10,10 @@ Game::Game(float currentHeight, float currentWidth)
 	player = Player();
 	player.UpdateFrameSize(height, width);
 	ResetLimiter();
-	asteroidCount = 5;
+	asteroidCount = 4;
```
Comment (rroa) [inline src/BoilerPlate/Game.cpp:13]: Move this to a constant. Avoid using magic numbers, only use literals when you're initializing something to a default value irrelevant to game logic.

File: src/BoilerPlate/Game.cpp [hunk 2]
```diff
@@ -55,25 +56,38 @@ void Game::Update(float currentHeight, float currentWidth, float deltaTime)
...
 void Game::Render()
 {
+	//===========================================================================
 	glClearColor(colors.slateBlue.redValue, colors.slateBlue.greenValue,
 		colors.slateBlue.blueValue, colors.slateBlue.alphaValue);
 	glClear(GL_COLOR_BUFFER_BIT);
+	//===========================================================================
 
 
+	//===========================================================================
```
Comment (rroa) [inline src/BoilerPlate/Game.cpp:66]: Honestly, don't like this "separators" but if you think they help the readability, go for it.

File: src/BoilerPlate/Game.cpp [hunk 3]
```diff
@@ -384,9 +398,11 @@ void Game::ManageInput()
 		ResetLimiter();
 	}
 
-	if (inputManager.GetR())
+	if (inputManager.GetR() && inputLimiter == 0)
```
Comment (rroa) [inline src/BoilerPlate/Game.cpp:401]: What's `GetR`? Is it state for the `R` key? Does it take into account if it's lower or upper case? Is it evaluating the state of the ship going `right`? Think about naming for this.

File: src/BoilerPlate/Game.cpp [hunk 4]
```diff
@@ -464,3 +480,26 @@ void Game::DrawFrameRateMeter()
...
+
+void Game::RenderLives()
+{
+	
+	lifePosition = 0;
+
+	for (int i = 0; i < playerLife; i++)
+	{
+		glLoadIdentity();
+		glTranslatef((-(width / 2.0f) + 50.0f) + lifePosition, ((height / 2.0f) - 50.0f), 0.0f);
```
Comment (rroa) [inline src/BoilerPlate/Game.cpp:492]: This offset translation is cool, but move the values to constant. Avoid magic numbers!

**Review by rroa — COMMENTED** [review]
Overall you've improved a lot in your coding style and design considerations. I'd try to be more consistent with naming, I still can't differentiate between private members and function local variables. Remember to be as explicit as possible with your function names.

**Comment by manuelmolinaf** [comment]
Ok, I'll get to this👍

**Pull request merged**